### PR TITLE
iOS 11.3 web support

### PR DESCRIPTION
**Pull request by imurchie**
Add support for iOS 11.3 mobile safari.

Also update Travis config:
* move to test against Xcode 9.2 with iOS 11.2
* move smoke tests to web tests, since these test the main functionality (WDA build, app start, etc) while also testing some of the more janky code
* add smoke tests (web) for Xcode 9.3 beta with iOS 11.3.

In the course of running Travis with Xcode 9.2, some issues came up with our tests' handling of simulators. So add some helper functions to not only shutdown simulators but clean up the xcodebuild processes, so we don't get multiple phantom sims running. This speeds up the tests and makes them more reliable.

### CODE REVIEW

File: lib/commands/web.js [hunk 1]
```diff
@@ -105,11 +113,17 @@ extensions.getExtraNativeWebTapOffset = async function () {
...
 extensions.nativeWebTap = async function (el) {
-  let atomsElement = this.useAtomsElement(el);
+  const atomsElement = this.useAtomsElement(el);
+
+  // `get_top_left_coordinates` returns the wrong value sometimes,
+  // unless we pre-call both of these functions before the actual calls
+  await this.executeAtom('get_size', [atomsElement]);
+  await this.executeAtom('get_top_left_coordinates', [atomsElement]);
+
+  const {width, height} = await this.executeAtom('get_size', [atomsElement]);
```
Comment (mykola-mokhnach) [inline lib/commands/web.js]: is there `getRect` primitive?

File: test/functional/driver/driver-e2e-specs.js [hunk 2]
```diff
@@ -29,105 +27,94 @@ const deleteDeviceWithRetry = async function (udid) {
...
           useNewWDA: true,
         });
         localCaps.wdaLocalPort = null;
-        await driver.init(localCaps);
+        driver = await initSession(localCaps);
         let logs = await driver.log('syslog');
-        if (!process.env.CI) {
-          logs.some((line) => line.message.indexOf(':8100<-') !== -1).should.be.true;
-        }
+        logs.some((line) => line.message.indexOf(':8100<-') !== -1).should.be.true;
```
Comment (mykola-mokhnach) [inline test/functional/driver/driver-e2e-specs.js]: includes?

File: test/functional/helpers/simulator.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import _ from 'lodash';
+import { getDevices, shutdown } from 'node-simctl';
+import { resetXCTestProcesses } from '../../../lib/utils';
+import { shutdownSimulator } from '../../../lib/simulator-management';
+import { killAllSimulators as simKill } from 'appium-ios-simulator';
+
+
+async function killAllSimulators () {
```
Comment (mykola-mokhnach) [inline test/functional/helpers/simulator.js]: why should we duplicate this in tests? I'd rather move the common code to utils.

File: test/functional/helpers/simulator.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const allDevices = _.flatMap(_.values(await getDevices()));
+  const bootedDevices = allDevices.filter((device) => device.state === 'Booted');
+
+  for (const {udid} of bootedDevices) {
+    // It is necessary to stop the corresponding xcodebuild process before killing
+    // the simulator, otherwise it will be automatically restarted
+    await resetXCTestProcesses(udid, true);
```
Comment (mykola-mokhnach) [inline test/functional/helpers/simulator.js]: shitdownSimulator ?

Comment (mykola-mokhnach) [reply test/functional/helpers/simulator.js]: `await B.forEach(bootedDevices, shutdownSimulator)`

File: lib/commands/web.js (hunk 1 above)
Comment (imurchie) [reply lib/commands/web.js]: No.

File: test/functional/helpers/simulator.js (hunk 4 above)
Comment (imurchie) [reply test/functional/helpers/simulator.js]: We have an array of `udid`s here, not simulator objects, so `shutdownSimulator` would not work.

Comment (jlipps) [reply test/functional/helpers/simulator.js]: love the Freudian slip there

**Pull request merged**